lib/Conversion/TritonToTritonGPU/TritonGPUConversion.cpp

```diff
@@ -79,7 +79,26 @@ TritonGPUConversionTarget::TritonGPUConversionTarget(
     MLIRContext &context, TritonGPUTypeConverter &typeConverter)
     : ConversionTarget(context) {
   // TODO: we should also verify ops of TritonGPUDialect
-  addLegalDialect<triton::gpu::TritonGPUDialect>();
+  addDynamicallyLegalDialect<triton::gpu::TritonGPUDialect>([&](Operation *op) {
+    // make sure every RankedTensorType operand has encoding
+    for (auto operandType : op->getOperandTypes()) {
+      if (auto rankedTensorType = dyn_cast<RankedTensorType>(operandType)) {
```

Comment:
Please also check against `MemDescType` and `TensorDescType` type.

Reply:
On the second thought let's omit this. We are currently assigning local tensors with layout in the frontend, so it likely hits a MemDesc/TensorDesc that has empty layout here.